apps/dcos_net/src/dcos_net_app.erl

```diff
@@ -7,7 +7,11 @@
 
 -behaviour(application).
 
+-ifdef(WINDOWS).
+-define(DEFAULT_CONFIG_DIR, os:getenv("DCOS_NET_CONFIG_DIR")).
+-else.
 -define(DEFAULT_CONFIG_DIR, "/opt/mesosphere/etc/dcos-net.config.d").
```

Comment:
We must make it configurable, but it's not a right place for environment variables. Erlang has its own logic for that. Please see the following links

[1] http://erlang.org/doc/man/config.html
[2] http://erlang.org/doc/man/erl.html#id200918
[3] http://erlang.org/doc/apps/kernel/application.html#get_env-2

I've created a separate JIRA for this issue, you can find it [here](https://jira.mesosphere.com/browse/DCOS_OSS-2308). Here is a PR: https://github.com/dcos/dcos-net/pull/39. I haven't tested it yet. The most challenging part is that VM calls this code even before many `kernel` and `stdlib` features initialized.

Reply:
Dropping the change. I'll test out the PR and provide as much feedback as I can.